spec/table_spec.rb

```diff
@@ -139,4 +139,136 @@
       eq(%{<table id='' class='data_table ' cellspacing='0' cellpadding='0'><caption></caption><thead><tr></tr></thead><tr><td class='empty_data_table' colspan='0'>#{text}</td></tr></table>})
     end
   end
+
+  context 'with a more complicated setup' do
+    it 'renders okay' do
+      raw_results = [
```

Comment:
It's probably not worth addressing this now but I'm wondering if we should put this test into a sub group and extract some of these setup items into `let` statements so we can get some reuse from them if we need to mock other complex scenarios.

Having said that they may all be too specific to get any reuse.

Reply:
> Having said that they may all be too specific to get any reuse.

That is my current thinking. This can be revisited if necessary, even as soon as another test is added.